configure.ac

```diff
@@ -36,6 +36,17 @@ AM_CONDITIONAL([MACOS], [test $macos -eq 1])
 AM_CONDITIONAL([LINUX], [test $linux -eq 1])
 AM_CONDITIONAL([FREEBSD], [test $freebsd -eq 1])
 
+AC_ARG_ENABLE([debug],
+	[AS_HELP_STRING([--enable-debug],
+			[Enable debugging - default NO])],
+	[CFLAGS="$CFLAGS -O0 -g -Werror"
```

Comment:
Why add -Werror? libfabric has other warning flags (Wall, Wextra, etc.) when debug mode is enabled. We could add them too.
Debug mode would be a good addition. :+1: